### ISSUE TITLE
Evaluate expression in debug mode cannot access lombok generated logger

### ISSUE THREAD

**Issue by daijithegeek** [mplushnikov/lombok-intellij-plugin#231]
Hello,

I created a class annotated with @Slf4j. During debug session, evaluating expressions containing java 8 streams fails.
![evaluate expression_048](https://cloud.githubusercontent.com/assets/2987252/15006758/a4eec1d8-11d3-11e6-9b28-c2405d520432.png)

**Comment by AlexejK** [comment]
Does non-java8 expression work for the class/method that contains auto-generated variable?

**Comment by daijithegeek** [comment]
yes the different cases I tested are working. You can even access the generated logger without any troubles. Apparently, only java8 or streaming API have issues

**Comment by AlexejK** [comment]
What IDEA version are you on? and what plugin version are you using?

**Comment by daijithegeek** [comment]
Sorry I forgot to mention that..
IDEA 15.0.5
lombok plugin: 0.11.15

**Comment by AlexejK** [comment]
Would it be possible for you to try newest IDEA version? 2016.1 improved a lot of things.
Also there are a few things in 0.12 that should possibly help out, but we can double-check if it's working as expected. 

Could you please provide a small sample project and exact steps to reproduce the issue, if 2016.1 is still not working for you?

**Comment by daijithegeek** [comment]
So, I've not a working licence for 2016.1 so it won't be a solution for me on the short term :/

Anyway I tried with idea 2016.1 and lombok plugin 0.11.16. Same bug appears. I'll provide you with a sample project today.

**Comment by AlexejK** [comment]
Thanks, I have a feeling this might be related to Modifier Visibility enhancements and one of the things that 2016.2 will introduce. However we have a work-in-progress backport for that to support IDEA 15 and 2016.1 as well.

But need to run it on a sample project to make sure.

**Comment by daijithegeek** [comment]
[my-test.tar.gz](https://github.com/mplushnikov/lombok-intellij-plugin/files/248652/my-test.tar.gz)

Here is a sample project. The error message is different than the initial screen shot, but I think the problem is related.

Steps to reproduce:

1. import the project
2. Add a breakpoint at line 16 of MyTest class 
3. Run MyTest main()  in debug mode until the BP
4. in Debug tab open the 'evaluate expression' window
5 enter expression: 'strings.stream().filter(s -> s.length() > 1).collect(Collectors.toSet())'
6. execute

**Comment by AlexejK** [comment]
Thanks, I was able to reproduce it. It seems that for some reason augmentation is not going the same way when evaluating expressions, even though I see it happening. We will have to dig deeper into this, even though it might be IDEA issue as well. 

If replacing `@Slf4j` with simple `private static final Logger log = LoggerFactory.getLogger(MyTest.class);` Evaluation does go through.

@mplushnikov: Any ideas?

**Comment by mplushnikov** [comment]
As long as I can see, IntelliJ uses [com.intellij.debugger.ui.impl.watch.CompilingEvaluatorImpl](https://github.com/JetBrains/intellij-community/blob/10a4c91dbfca935b4b5531003fd8c13b56e66202/java/debugger/impl/src/com/intellij/debugger/ui/impl/watch/CompilingEvaluatorImpl.java) to compile dynamically created class with some code for evaluation-string. 
But the compiler is started with annotation processiing disabled:
```java
options.add("-proc:none"); 
```
 and is unable to undestand code with lombok extentions.

I'll create an issue for Jetbrains to change it.

**Comment by mplushnikov** [comment]
[IDEA-156045](https://youtrack.jetbrains.com/issue/IDEA-156045) Evaluate expression in debug mode cannot access any of lombok code extensions

**Comment by mplushnikov** [comment]
The current EAP Build of Intellij 2016.2 should be much better in such situations. Unfortunately there are still some minor with specific expression evaluations. 
I'll continue to investigate it...

**Comment by yannxia** [comment]
Intellij  2017.1.3  this is still a question .....

**Comment by kerraway** [comment]
IntelliJ IDEA 2017.3.5 (Ultimate Edition)
Build #IU-173.4674.33, built on March 6, 2018

This issues still exists.

**Comment by afwlehmann** [comment]
IntelliJ IDEA 2019.1.3 (Ultimate Edition)
Build #IU-191.7479.19, built on May 28, 2019

Issue still open.

**Comment by angelstar02** [comment]
IntelliJ IDEA 2020.1 (Ultimate Edition)
Build #IU-201.6668.121, built on April 8, 2020

Issue still open
